config/examples/vmservicescrape-metrics-server.yml

```diff
@@ -0,0 +1,53 @@
+# Scrape Metrics Server
+# Cluster-wide aggregator of resource usage data. 
+# https://github.com/kubernetes-sigs/metrics-server
+---
+apiVersion: operator.victoriametrics.com/v1beta1
+kind: VMServiceScrape
+metadata:
+  name: metrics-server
+  namespace: kube-system
+spec:
+  namespaceSelector:
+    matchName:
+    - kube-system
```

Comment:
@f41gh7 I'm not sure about this as changing this value doesn't seems to have any effect to resulting configuration, it's defined in CRD but is it really doing anything?

Reply:
Well, there is typo - it must be `matchNames` instead of `matchName` :)

And in this case, yes, it does nothing. Because  namespace selector  was added implicitly, if selectors is not defined.
 There is complicated logic around it:
1) spec.ignoreNamespaceSelectors = true - crd namespace will be added.
2) if spec.namespaceSelector.any = true - empty selector, match all namespaces
3) if len(matchNames) == 0 - crd namespace will be added
4) otherwise use user defined list of namespaces.

Reply:
Ah silly mistake 😄 thanks! I have fix it and place VMServiceScrape to the `monitoring-system`, so others have guides how to target other namespaces. It would be great if for each object there was example file with all options with inline comments, ideally automatically generated from CRD definition as browsing thru definition itself isn't much fun 😉.

Also this leads me to question about debug mode, because I had to come up with technique ( `kubectl exec -it -n monitoring-system vmagent-vmagent-6d4ffc5697-mzxr9 -c config-reloader -- gunzip /etc/vmagent/config/vmagent.yaml.gz -c` ) to figure out what is generated in order to make this manifests to work and it would be much faster if I could turn vmagent into debug mode where it would print it's config on init and every time it's changed, wdyt?

Reply:
Yep, i faced the same problem. I thought about code-generator for such examples, but its tricky to write it, there are fields with two options (like intOrStr or Configmap or Secret and so on). It will take some time to write it, probably take a look this week.
 
   Unfortunately no, operator-sdk logger has only Info and Error methods, it possible to replace it with some other logger, but i dont think, that its an option for now. I suggest to use following command:

 ```bash
 kubectl -n monitoring-system get secrets vmagent-CRD_NAME -o 'go-template={{index .data "vmagent.yaml.gz"}}' | base64 -d | gunzip
```
 with it you can use yq for filtering https://github.com/mikefarah/yq